src/app/home/layouts/budgets-form/budgets-form.component.html

```diff
@@ -7,56 +7,64 @@ <h2 class="header">Budgets</h2>
     <i class="fas fa-plus-circle add-icon" (click)="addBudget()"></i>
   </div>
 
-  <div *ngIf="isNewBudgetCardVisible" class="budget-form-container" @fade>
+  <div *ngIf="isNewBudgetCardVisible" class="budget-form-container">
 
     <div class="x-icon-container">
       <i class="fas fa-times x-icon" (click)="isNewBudgetCardVisible = false"></i>
     </div>
 
-    <fieldset class="budget-fieldset">
-      <text-input name="Name"
-                  ngModel
-                  required></text-input>
+    <form #budgetForm="ngForm" (ngSubmit)="saveBudget(budgetForm)" @fade>
 
-      <text-input name="Amount"
-                  ngModel
-                  type="number"
-                  [isCheckIconVisible]="false"
-                  required></text-input>
+      <fieldset class="budget-fieldset">
+        <text-input name="Name"
+                    ngModel
+                    required></text-input>
 
-      <text-input name="Date"
-                  ngModel
-                  type="number"
-                  [isCheckIconVisible]="false"
-                  required></text-input>
-    </fieldset>
+        <text-input name="Amount"
+                    ngModel
+                    type="number"
+                    [isCheckIconVisible]="false"
+                    required></text-input>
 
-    <p class="cycle-header">Cycle</p>
-    <div class="cycle-container">
-      <div class="cycle-element">Weekly</div>
-      <div class="cycle-element">Monthly</div>
-      <div class="cycle-element">Yearly</div>
-    </div>
+        <text-input name="Date"
+                    ngModel
+                    type="number"
+                    [isCheckIconVisible]="false"
+                    required></text-input>
+      </fieldset>
 
-    <h4 class="category-header">Category</h4>
-    <div class="categories-container">
-      <div *ngFor="let category of categories">
-        <i class="fas fa-circle circle"></i>
-        <p class="category-text">{{category}}</p>
+      <p class="cycle-header">Cycle</p>
+      <div class="cycle-container">
+        <div *ngFor="let cycle of cycles"
+             [class]="cycle.isSelected ? 'cycle-element selected-cycle' : 'cycle-element standard-cycle'"
+             (click)="selectCycle(cycle)">{{cycle.name}}
+        </div>
       </div>
-      <div class="category-add-element">
-        <i class="fas fa-circle circle"></i>
-        <i class="fas fa-plus add-category-icon"></i>
+
+      <h4 class="category-header">Category</h4>
+      <div class="categories-container">
+
+        <div *ngFor="let category of categories">
+          <i [class]="category.isSelected ?
+          'fas fa-circle category-circle selected-category-color' :
+          'fas fa-circle category-circle standard-category-color'"
```

Comment:
Do not repeat common classes  for each condition. In the ternary expression, bind only classes that differ, the rest just put as a separate un-bound class property.